src/AbstractArray.php

```diff
@@ -180,7 +180,7 @@ public function __construct(array $elements = [])
      *
      * @return static The instance array with combined elements
      */
-    abstract public function combineTo(array $array);
+    abstract public function useAndCombineTo(array $array);
```

Comment:
@nyamsprod Well, lets get `useAndReplaceIn` method for example. Actually, this method is truly do 2 actions:
* First, **use** passed array as a main one.
* And replace a new passed array with current (old) one.

I don't sure you completely understand purpose of this 2 methods. Let me explain a bit. To be concise:
* `$current->replaceWith($new)` is equal to `array_replace($current, $new)`.
* `$current->useAndReplaceIn($new)` is equal to `array_replace($new, $current)`. It's an inversed side of first case, i.e. I transform this new array to the Arrayzy object first and then replace in it the old one.

So what about `replaceKeysWith` and `replaceValuesWith` methods, based on their names, I think now it's seen that they're doing something different tasks.

I hope I was able to explain it well to you.